private/pkg/protosource/field.go

```diff
@@ -107,8 +107,40 @@ func newField(
 	}
 }
 
-func (f *field) Message() Message {
-	return f.message
+func (f *field) ParentMessage() Message {
+	return f.parentMessage
+}
+
+func (f *field) Embed(files ...File) Message {
```

Comment:
Why is this public? This function worries me - why are we taking a list of `Files` into a `Field`? This feels suceptible to user error.

I'd recommend we use the solution we discussed offline and not add this functionality in this way to `protosource`.

Reply:
I think I've misunderstood our offline discussion - let me close out everything else and come back to this point.

Reply:
```func IsFieldAMap(field Field, fullNameToMessage map[string]Message) bool {
	// For this to be a map field, it must be a repeated field
	// with a synthetic message as the type, that synthetic message
	// must be in the enclosing message for the field, and it must
	// have map_entry set to true.
	if field.Label() != descriptorpb.FieldDescriptorProto_LABEL_REPEATED {
		return false
	}
	if field.Type() != descriptorpb.FieldDescriptorProto_TYPE_MESSAGE {
		return false
	}
	message, ok := fullNameToMessage[field.TypeName()]
	if !ok {
		return false
	}
	return message.IsMapEntry()
}```

Reply:
resolved in [ae405eb](https://github.com/bufbuild/buf/pull/2377/commits/ae405eb097ab57139d9f22eea4ec346740c1b3ea)